src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/TerraformScriptsService.java

```diff
@@ -0,0 +1,165 @@
+            log.error("Deployment service error, taskId: {} , error: {}", request.getTaskId(),
+                    e.getMessage());
+            return TerraformResult.builder()
+                    .commandStdOutput(null)
+                    .commandStdError(e.getMessage())
+                    .isCommandSuccessful(false)
+                    .terraformState(null)
+                    .build();
+        }
+    }
+
+    /**
+     * Method of destroy a service using a script.
+     */
+    public TerraformResult destroyWithScripts(TerraformDestroyWithScriptsRequest request) {
+        buildDestroyEnv(request.getTaskId(), request.getScripts(), request.getTfState());
+        return destroyFromDirectory(request, request.getTaskId());
+    }
+
+    /**
+     * Async deploy a source by terraform.
+     */
+    @Async("taskExecutor")
+    public void asyncDeployWithScripts(
+            TerraformAsyncDeployFromDirectoryRequest asyncDeployRequest) {
+        TerraformResult result = deployWithScripts(asyncDeployRequest);
+        log.info("Deployment service complete, {}", result.getCommandStdOutput());
+        restTemplate.postForLocation(asyncDeployRequest.getWebhookConfig().getUrl(), result);
+    }
+
+    /**
+     * Async destroy resource of the service.
+     */
+    @Async("taskExecutor")
+    public void asyncDestroyWithScripts(TerraformAsyncDestroyFromDirectoryRequest request) {
+        TerraformResult result = destroyWithScripts(request);
+        log.info("Destroy service complete, {}", result.getCommandStdOutput());
+        restTemplate.postForLocation(request.getWebhookConfig().getUrl(), result);
+    }
+
+    private String buildDeployEnv(String taskId, List<String> scripts)
+            throws JsonProcessingException {
+        String workspace = executor.getModuleFullPath(taskId);
+        buildWorkspace(workspace);
+        buildScriptFile(workspace, scripts);
+        return workspace;
+    }
+
+    private String buildDestroyEnv(String taskId, List<String> scripts, String tfState) {
+        String workspace = executor.getModuleFullPath(taskId);
+        buildScriptFile(workspace, scripts);
+
+        if (StringUtils.isBlank(tfState)) {
+            throw new TerraformExecutorException("terraform .tfState file create error");
+        }
+        String fileName = workspace + File.separator + STATE_FILE_NAME;
+        try (FileWriter scriptWriter = new FileWriter(fileName)) {
+            scriptWriter.write(tfState);
+            log.info("terraform .tfState file create success, fileName: {}", fileName);
+        } catch (IOException ex) {
+            log.error("terraform .tfState file create failed.", ex);
+            throw new TerraformExecutorException("terraform .tfState file create failed.", ex);
+        }
+        return workspace;
+    }
+
+    /**
+     * Build workspace of the `terraform`.
+     *
+     * @param workspace The workspace of the task.
+     */
+    private void buildWorkspace(String workspace) {
```

Comment:
in case of scripts -> the workspace name must be a random too. No dependency on any other task id or anything else